tunnel/listener.go

```diff
@@ -17,26 +18,81 @@ type tunListener struct {
 	tunClosed chan bool
 	// the listener session
 	session *session
+	// del func to kill listener
+	delFunc func()
+}
+
+// periodically announce self
+func (t *tunListener) announce() {
+	tick := time.NewTicker(time.Minute)
+	defer tick.Stop()
+
+	announce := func() {
```

Comment:
Wouldnt turning this `func` definition into a method be "nicer"? If defined as method it's defined once and possibly testable should we decide to start writing tests at some point?

Reply:
Moved into session as an Announce method.